doc/sphinx/language/core/inductive.rst

```diff
@@ -13,7 +13,7 @@ Inductive types
 
    .. prodn::
       inductive_definition ::= @ident {? @cumul_univ_decl } {* @binder } {? %| {* @binder } } {? : @type } := {? %| } {+| @constructor } {? @decl_notations }
-      constructor ::= @ident {* @binder } {? @of_type_inst }
+      constructor ::= {* #[ {*, @attribute } ] } @ident {* @binder } {? @of_type_inst }
```

Comment:
Probably but before that we should first describe the initial syntax. I'm not sure the feature is used much though, it seems it was mostly introduced for definitional typeclasses (typeclasses with a single field and no constructor). Reusing the constructor syntax for that was probably a bad idea by the way but well, now it is. Fun fact, the attribute works fine for definitional typeclasses but is ignored for actual constructors, this thing is really a mess.
BTW, since variant comes before inductive, maybe this should be moved there (otherwise there is a forward link in variant).

Reply:
>since variant comes before inductive, maybe this should be moved there (otherwise there is a forward link in variant

:+1:

>Fun fact, the attribute works fine for definitional typeclasses but is ignored for actual constructors

Not so fun. It would be nicer to get an error message in this case.

>Probably but before that we should first describe the initial syntax. I'm not sure the feature is used much though, it seems it was mostly introduced for definitional typeclasses (typeclasses with a single field and no constructor). Reusing the constructor syntax for that was probably a bad idea by the way but well, now it is.

One way to document this would be to add a description of the feature in the typeclass chapter and to forward-reference it when describing why this syntax is defined here.

Reply:
> Not so fun. It would be nicer to get an error message in this case.

Sorry, I was overly ironic. I'll certainly fix it at some point but honestly this is not very high on my todolist.